pkgpanda/util.py

```diff
@@ -167,14 +200,7 @@ def download(out_filename, url, work_dir, rm_on_error=True):
                 src_filename = work_dir + '/' + src_filename
             shutil.copyfile(src_filename, out_filename)
         else:
-            # Download the file.
-            with open(out_filename, "w+b") as f:
-                r = get_requests_retry_session().get(url, stream=True)
-                if r.status_code == 301:
-                    raise Exception("got a 301")
-                r.raise_for_status()
-                for chunk in r.iter_content(chunk_size=4096):
-                    f.write(chunk)
+            stream_remote_file_with_retries(out_filename, url)
```

Comment:
I think `_with_retries` is an implementation detail, let's just call the function "_download_file" ?

Reply:
I wanted to differentiate it from the existing `download` function and make its possibly slow behaviour explicit, but I get your point.

I'll rename it to `_download_remote_file`, since `download` has two branches: "download" local file or download remote file. The latter is what's encapsulated in this function.